gef.py

```diff
@@ -10737,8 +10737,22 @@ def read_cstring(self,
         try:
             res_bytes = self.read(address, length)
         except gdb.error:
-            err(f"Can't read memory at '{address}'")
-            return ""
+            current_address = address
+            res_bytes = b""
+            while len(res_bytes) < length:
+                try:
+                    new_end_addr = current_address + DEFAULT_PAGE_SIZE
+                    page_mask = ~(DEFAULT_PAGE_SIZE - 1)
+                    size = (new_end_addr & page_mask) - current_address
```

Comment:
comment something like "calculate size by going to the next page-aligned address after current address"